draft-ietf-httpbis-safe-method-w-body.xml

```diff
@@ -285,27 +285,35 @@ q=foo&amp;limit=10&amp;sort=-published
 
   <section title="The &quot;Accept-Query&quot; Header Field" anchor="field.accept-query">
     <t>
-      The "Accept-Query" response header field &MAY; be used by a resource to
+      The "Accept-Query" response header field can be used by a resource to
       directly signal support for the QUERY method while identifying
       the specific query format media type(s) that may be used.
     </t>
-<sourcecode type="abnf">
-Accept-Query = 1#media-type
-</sourcecode>
     <t>
-      The Accept-Query header field specifies a comma-separated listing of media
-      types (with optional parameters) as defined by
-      <xref target="HTTP" section="8.3.1"/>. <cref>field syntax currently discussed in <eref target="https://github.com/httpwg/http-extensions/issues/2860"/></cref>
+      "Accept-Query" contains a list of media types (<xref target="HTTP" section="8.3.1"/>),
+      represented by a List Structured Header Field, containing
+      Token Items, optionally parameterized (<xref target="STRUCTURED-FIELDS" section="3"/>).
```

Comment:
What is/are the type(s) of the parameters?

Reply:
the `parameter-value` of a `media-type` or `media-range` can be a `token` or a `quoted-string`. in order to express media types and their parameters in a SF as commonly given, both token and quoted string need to be supported. quoted string is definitely needed because some media types' parameters (can) have spaces in them that mean something (for example, the `profile` parameter of `application/ld+json`).

Reply:
Though not likely useful, `q=0.9` would be sf-decimal .

Reply:
@martinthomson - I believe @mnot prefers to explicitly restrict the allowed types to the bare minimum.